BUILDING.md

```diff
@@ -1,11 +1,51 @@
+# Build-System CLI
+
+## Non-interactive
+```
+python build.py -h, --help
+
+usage: build.py [-h] --target {android,linux,macos,win32} --arch {x86,x64,arm}
+                [--vendor VENDOR] [--keep-native-libs] [--node-enabled]
+                [--docker] [--vagrant] [--sys-image SYS_IMAGE] [--no-shutdown]
+                [--interactive]
+                [build-steps [build-steps ...]]
+```
+```
+python build.py -v alpine -t linux -a x64 -dkr -img openjdk:8u131-alpine -ne j2v8
+```
+
+## Interactive
+```
+python build.py --i, --interactive
+
+entering interactive mode...
+
+[0] Docker >> android-x86 >> NODE_ENABLED
+[1] Docker >> android-arm >> NODE_ENABLED
+[2] Docker >> alpine-linux-x64 >> NODE_ENABLED
+[3] Docker >> linux-x64 >> NODE_ENABLED
+[4] Docker >> linux-x86 >> NODE_ENABLED
+[5] Vagrant >> macosx-x64 >> NODE_ENABLED
+[6] Vagrant >> macosx-x86 >> NODE_ENABLED
+[7] Native >> windows-x64 >> NODE_ENABLED
+[8] Docker >> windows-x64 >> NODE_ENABLED
+[9] Vagrant >> windows-x64 >> NODE_ENABLED
+
+Select a predefined build-configuration to run: 2
+Building: Docker >> alpine-linux-x64 >> NODE_ENABLED
+
+Override build-steps ? (leave empty to run pre-configured steps): j2v8
+```
+
 # Build-Steps
 
 The J2V8 build-system performs several build steps in a fixed order to produce the final J2V8 packages for usage on the designated target platforms. What follows is a short summary for what each of the executed build-steps does and what output artifacts are produced by each step.
 
 ---
 ## Node.js
 
-Builds the [Node.js](https://nodejs.org/en/) & [V8](https://developers.google.com/v8/) dependency artifacts that are later linked against by the J2V8 native bridge code.
+Builds the [Node.js](https://nodejs.org/en/) & [V8](https://developers.google.com/v8/) dependency artifacts that are later linked into the J2V8 native bridge code.
```

Comment:
This script is far from finished, I need to first get a better picture of how this would be used in a CI build before I start this kind of polishing :wink: